src/test/com/opinta/controller/ShipmentIntegrationTest.java

```diff
@@ -0,0 +1,171 @@
+                .path("id");
+
+        expect()
+                .statusCode(SC_OK)
+                .when()
+                .get("shipments/{id}", newShipmentId);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void updateShipment() throws Exception {
+        JSONObject shipment = new JSONObject();
+        shipment.put("senderId", 1);
+        shipment.put("recipientId", 2);
+        shipment.put("deliveryType", "D2D");
+        shipment.put("weight", 10);
+        shipment.put("length", 12);
+        shipment.put("width", 0);
+        shipment.put("height", 0);
+        shipment.put("declaredPrice", 24);
+        shipment.put("price", 3.5);
+        shipment.put("postPay", 84);
+        shipment.put("description", null);
+
+        given()
+                .contentType("application/json")
+                .body(shipment.toJSONString())
+                .expect()
+                .statusCode(SC_OK)
+                .when()
+                .put("/shipments/{id}", 1)
+                .then()
+                .body("id", greaterThan(0));
+
+        expect()
+                .statusCode(SC_OK)
+                .when()
+                .get("shipments/{id}", 1)
+                .then()
+                .body("senderId", equalTo(1))
+                .body("recipientId", equalTo(2))
+                .body("deliveryType", equalTo("D2D"));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void deleteShipment() throws Exception {
+        try {
+            expect()
+                    .statusCode(SC_OK)
+                    .when()
+                    .delete("shipments/{id}", 1);
+        } catch (AssertionError e) {
+            JSONObject shipment = new JSONObject();
+            shipment.put("senderId", 1);
+            shipment.put("recipientId", 2);
+            shipment.put("deliveryType", "D2D");
+            shipment.put("weight", 10);
+            shipment.put("length", 12);
+            shipment.put("width", 0);
+            shipment.put("height", 0);
+            shipment.put("declaredPrice", 24);
+            shipment.put("price", 3.5);
+            shipment.put("postPay", 84);
+            shipment.put("description", null);
+
+            int newShipmentId = given()
+                    .contentType("application/json")
+                    .body(shipment.toJSONString())
+                    .expect()
+                    .statusCode(SC_OK)
+                    .when()
+                    .post("/shipments")
+                    .then()
+                    .body("id", greaterThan(0))
+                    .extract()
+                    .path("id");
+
+            expect()
+                    .statusCode(SC_OK)
+                    .when()
+                    .delete("shipments/{id}", newShipmentId);
+
+            expect()
+                    .statusCode(SC_NOT_FOUND)
+                    .when()
+                    .get("shipments/{id}", newShipmentId);
+        }
+    }
+}
```

Comment:
This class is absolutely fine, but if would cause 'mvn clean package' to fail.

The point is that this integration test is placed under src/test in its natural package. It will be invoked before the application will be launced, though rest assured will fail as it will not be able to setup http connection with unlaunced server instance.

Please, checkout the pom.xml and src/test subfolders in my branch related to integration tests to see how integration tests are separated from unit test in order 'mvn clean package' execution invokes only unit, but not integration tests.